react/Modal/index.jsx

```diff
@@ -3,57 +3,98 @@ import classNames from 'classnames'
 
 import styles from './index.styl'
 
-const Modal = ({
-  title, description,
-  cancelType, cancelText, cancelAction,
-  validateType, validateText, validateAction, children
-}) => (
-  <div className={styles['coz-modal-container']}>
-    <div className={styles['coz-overlay']}>
-      <div className={styles['coz-modal']}>
-        <h2 className={styles['coz-modal-title']}>{title}</h2>
-        <button
-          className={classNames('coz-btn', 'coz-btn--close', styles['coz-btn-modal-close'])}
-          onClick={cancelAction}
-          >
-          <span className='coz-hidden'>{cancelText}</span>
-        </button>
-        { description && <div className={classNames(styles['coz-modal-content'], styles['coz-description'])}>
-            {description}
-          </div>
+const ModalTitle = ({ title }) =>
+  (
+    <h2 className={styles['coz-modal-title']}>{title}</h2>
+  )
+
+const ModalCross = ({ withCross, secondaryAction, secondaryText }) =>
+  withCross &&
+  (
+    <button
+      className={classNames('coz-btn', 'coz-btn--close', styles['coz-btn-modal-close'])}
+      onClick={secondaryAction}
+      >
+      <span className='coz-hidden'>{secondaryText}</span>
+    </button>
+)
+
+const ModalDescription = ({ description }) =>
+  description &&
+  (
+    <div className={classNames(styles['coz-modal-content'], styles['coz-description'])}>
+      {description}
+    </div>
+  )
+
+const ModalButtons = ({ secondaryText, secondaryAction, secondaryType, primaryText, primaryAction, primaryType }) => {
+  const displayPrimary = primaryText && primaryAction
+  const displaySecondary = secondaryText && secondaryAction
+  return (displaySecondary || displayPrimary) &&
+    (
+      <div className={classNames(styles['coz-modal-content'], styles['coz-modal-buttons'])}>
+        { displaySecondary &&
+          <button className={classNames('coz-btn', 'coz-btn--' + secondaryType)} onClick={secondaryAction}>
+            {secondaryText}
+          </button>
         }
-        { children }
-        { ((cancelText && cancelAction) || (validateText && validateAction)) &&
-          <div className={classNames(styles['coz-modal-content'], styles['coz-modal-buttons'])}>
-            { (cancelText && cancelAction) && <button className={classNames('coz-btn', 'coz-btn--' + cancelType)} onClick={cancelAction}>
-              {cancelText}
-            </button>
-            }
-            { (validateText && validateAction) && <button className={classNames('coz-btn', 'coz-btn--' + validateType)} onClick={validateAction}>
-              {validateText}
-            </button>
-            }
-          </div>
+        { displayPrimary &&
+          <button className={classNames('coz-btn', 'coz-btn--' + primaryType)} onClick={primaryAction}>
+            {primaryText}
+          </button>
         }
       </div>
+    )
+}
+
+const Modal = (props) => {
+  props.primaryType = props.primaryType || props.validateType
```

Comment:
I do not use destructuring here because I think:

```javascript
const Modal = ({ title, description, primaryType, primaryText, primaryAction, secondaryType, secondaryText, secondaryAction, validateType, validateText, validateAction, cancelType, cancelText, cancelAction, withCross }) => { /* etc. */ }
```

was less clean than 

```javascript
const Modal = (props) => { /* etc. */ }
```

mostly when I decomposed the massive Modal component into smaller one by passing them `...props` instead of the props they need explicitly. Especially with a component that declares its propTypes.

But I may be wrong.

What do you think about?